core/src/main/java/org/jdbi/v3/core/statement/StatementContext.java

```diff
@@ -67,6 +68,7 @@
  */
 public class StatementContext implements Closeable {
 
+    private final Class<?> jdbiStatementType;
```

Comment:
is that just for illustration purposes or does it have to be a Class<?>? If the former, can we make this a Type? We have been burned a few times by using Class<?> and then it turns out that Type would have been a better choice.

Reply:
Not important, just wanted a way to differentiate query vs update.
(Another user asked for this too for their own reasons, so it is intentional to make it public)

Reply:
I tried to change this and it makes things a bit more complicated. I want to call `Class.getSimpleName` which is not on the `Type` interface. So, to handle `Type` here, we need to special case back to the Class we already have... if you think this is better though it's ok by me

Reply:
I like simpler. It is marked as `@Alpha` anyway. Let's leave it like this.